third_party/move/move-vm/runtime/src/storage/module_storage.rs

```diff
@@ -472,10 +460,24 @@ impl<'a> FunctionValueExtension for FunctionValueExtensionAdapter<'a> {
                         ty_builder.create_ty_with_subst(ty, &fun.ty_args)
                     }
                 };
+
+                let loader = NativeLoader::new(self.module_storage);
+                let layout_converter = LayoutConverter::new(&loader);
+
                 let captured_layouts = mask
                     .extract(fun.param_tys(), true)
                     .into_iter()
-                    .map(|t| ty_converter.type_to_type_layout(&instantiate(t)?))
+                    .map(|t| {
+                        // TODO(georgemitenkov):
+                        //   Double check this is correct and aggregators work with closures.
```

Comment:
Is this coming in later PRs? What's the simplest possible test we can have for it / integration pieces?

Reply:
This is unrelated to lazy loading, will create function values aggregator tests separately to test this

Reply:
got it, I hope we can eventually get to a place where things can be tested below e2e tests.. But that's probably (a few++) refactors away lol